core/src/main/java/com/github/jsonldjava/utils/JarCacheStorage.java

```diff
@@ -126,13 +125,28 @@ public HttpCacheEntry getEntry(String key) throws IOException {
         return delegate.getEntry(key);
     }
 
-    private Enumeration<URL> getResources() throws IOException {
-        final ClassLoader cl = getClassLoader();
-        if (cl != null) {
-            return cl.getResources(JARCACHE_JSON);
-        } else {
-            return ClassLoader.getSystemResources(JARCACHE_JSON);
+    /**
+     * Get all of the {@code jarcache.json} resources that exist on the classpath
+     * 
+     * @return A cached list of jarcache.json classpath resources as {@link URL}s
+     * @throws IOException If there was an IO error while scanning the classpath
+     */
+    private List<URL> getResources() throws IOException {
+        List<URL> result = cachedResourceList;
+        if(result == null) {
```

Comment:
If you want to preserve the previous behavior of allowing the switching of the Thread.contextClassLoader(), you could always add an additional field `ClassLoader loaderUsedToLoad`, which you would set to the ClassLoader which loaded `cachedResourceList` previously, then check:

`if (result == null || loaderUsedToLoad != getClassLoader()) {...`